src/algorithm/jde.cpp

```diff
@@ -187,40 +187,78 @@ void jde::evolve(population &pop) const
 			}
 		}
 
+		std::vector<size_t> elements;
+		if (NP < 21) {
+			// Add elements in order.
+			for (size_t i = 0; i < NP; ++i) {
+				elements.push_back(i);
+			}
+		}
+
 		//Start of the loop through the deme
 		for (size_t i = 0; i < NP; ++i) {
-			do {                       /* Pick a random population member */
-				/* Endless loop for NP < 2 !!!     */
-				r1 = p_idx();
-			} while (r1==i);
-
-			do {                       /* Pick a random population member */
-				/* Endless loop for NP < 3 !!!     */
-				r2 = p_idx();
-			} while ((r2==i) || (r2==r1));
-
-			do {                       /* Pick a random population member */
-				/* Endless loop for NP < 4 !!!     */
-				r3 = p_idx();
-			} while ((r3==i) || (r3==r1) || (r3==r2));
-
-			do {                       /* Pick a random population member */
-				/* Endless loop for NP < 5 !!!     */
-				r4 = p_idx();
-			} while ((r4==i) || (r4==r1) || (r4==r2) || (r4==r3));
-
-			do {                       /* Pick a random population member */
-				/* Endless loop for NP < 6 !!!     */
-				r5 = p_idx();
-			} while ((r5==i) || (r5==r1) || (r5==r2) || (r5==r3) || (r5==r4));
-			do {                       /* Pick a random population member */
-				/* Endless loop for NP < 7 !!!     */
-				r6 = p_idx();
-			} while ((r6==i) || (r6==r1) || (r6==r2) || (r6==r3) || (r6==r4) || (r6==r5));
-			do {                       /* Pick a random population member */
-				/* Endless loop for NP < 8 !!!     */
-				r7 = p_idx();
-			} while ((r7==i) || (r7==r1) || (r7==r2) || (r7==r3) || (r7==r4) || (r7==r5) || (r7==r6));
+			if (NP < 21) {
```

Comment:
Currently we include everything in the evolve() method. Would it be a good idea to separate the sampling algorithm (both for small and large NP) into a separate function?